IanGRap/Squish
# Refactor Controller Input

Contributors: Austin Kellar

The Story:
- When we first migrated to CarbonInput, I did not know that there was built-in functionality to get the exact frame that a button was pressed down, so I was doing it manually by storing the button's state in the previous frame. I discovered a week ago that CarbonInput has something called GamePadState, which we can use to easily tell if a controller button was pressed down during a particular frame. GamePadState also has the benefit of capturing the state of all buttons in one go, rather than polling each individual button every frame.

Goals:
- Optimize the process of reading controller input by capturing the GamePadState only once per frame, rather than once per frame for each button
- Reduce the logic for ButtonDown events.

Important Files Changed:
- PlayerInput.cs
- PauseMenuInput.cs
- MatchManager.cs

How to Test:
- Load up any match scene (I used 2017) and make sure that all player movement input still works, as well as the pause menu.
- I am hoping that this will also solve the problems we have been having with the triggers and player dash. (fingers crossed)
